test/support/route.go

```diff
@@ -36,3 +40,45 @@ func GetRoute(t Test, namespace, name string) *routev1.Route {
 	t.T().Helper()
 	return Route(t, namespace, name)(t)
 }
+
+func ExposeService(t Test, namespace string, serviceName string, routeName string, servicePort string) *routev1.Route {
+	r := &routev1.Route{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: routev1.SchemeGroupVersion.String(),
+			Kind:       "Route",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      routeName,
+			Namespace: namespace,
+		},
+		Spec: routev1.RouteSpec{
+			To: routev1.RouteTargetReference{
+				Name: serviceName,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromString(servicePort),
+			},
+		},
+	}
+
+	_, err := t.Client().Route().RouteV1().Routes(r.Namespace).Create(t.Ctx(), r, metav1.CreateOptions{})
+	t.Expect(err).NotTo(gomega.HaveOccurred())
+	t.T().Logf("Created Route %s/%s successfully", r.Namespace, r.Name)
+
+	t.T().Logf("Waiting for Route %s/%s to be available", r.Namespace, r.Name)
+	t.Eventually(Route(t, r.Namespace, r.Name), TestTimeoutLong).
+		Should(gomega.WithTransform(ConditionStatus(routev1.RouteAdmitted), gomega.Equal(corev1.ConditionTrue)))
+
+	// Retrieve hostname
+	r, err = t.Client().Route().RouteV1().Routes(r.Namespace).Get(t.Ctx(), r.Name, metav1.GetOptions{})
+	t.Expect(err).NotTo(gomega.HaveOccurred())
+	hostname := r.Status.Ingress[0].Host
+
+	// Wait for expected HTTP code
+	t.Eventually(func() int {
```

Comment:
Can you also check that response body contains text specific to the Route not serving yet, for example:
`Route and path matches, but all pods are down.`

Reply:
The reason is that theoretically the 503 can be a correct response code for the service.
Though that may be really edge case.

Reply:
Currently not needed.